### PR TITLE
Add extended globs support for root_file

### PR DESCRIPTION
**Pull request by marcovolpato00**
This PR enables extended globbing for `root_file` (if `glob_root_file` is set). This is useful, for example, when you want to ignore some directories. As described [here](https://unix.stackexchange.com/a/276923), this feature should be safe.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -118,8 +125,12 @@ jobs:
           file test/file2.pdf | grep -q ' PDF '
           file test/glob_test1.pdf | grep -q ' PDF '
           file test/glob_test2.pdf | grep -q ' PDF '
+          file test/glob_test2.pdf | grep -q ' PDF '
```
Comment (xu-cheng) [inline .github/workflows/test.yml:128]: Duplicated line?

File: entrypoint.sh [hunk 2]
```diff
@@ -44,6 +44,7 @@ if [[ -n "$working_directory" ]]; then
 fi
 
 if [[ -n "$glob_root_file" ]]; then
+  shopt -s extglob
```
Comment (xu-cheng) [inline entrypoint.sh:47]: Could we simply enable this flag globally?

**Review by xu-cheng — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (marcovolpato00) [reply .github/workflows/test.yml]: Yup, fix pushed

File: entrypoint.sh (hunk 2 above)
Comment (marcovolpato00) [reply entrypoint.sh]: Yes absolutely, I was afraid of breaking stuff by enabling the flag globally but it shouldn't

**Pull request merged**